src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -232,6 +232,7 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
         _innerBuilder.Services.AddSingleton(options);
         _innerBuilder.Services.AddSingleton<ResourceNotificationService>();
         _innerBuilder.Services.AddSingleton<ResourceLoggerService>();
+        _innerBuilder.Services.AddSingleton<ResourceCommandService>(s => new ResourceCommandService(s.GetRequiredService<ResourceNotificationService>(), s.GetRequiredService<ResourceLoggerService>(), s));
```

Comment:
Why is this so complex?

```suggestion
        _innerBuilder.Services.AddSingleton<ResourceCommandService>();
```


Should resolve just fine.

Reply:
It's required because the constructor is internal.